src/Domains/Connectors/Shopify/ShopifyService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Connectors\Shopify;
+
+use Baka\Contracts\AppInterface;
+use Baka\Contracts\CompanyInterface;
+use Kanvas\Connectors\Shopify\DataTransferObject\Shopify as ShopifyDto;
+use Kanvas\Connectors\Shopify\Enums\CustomFieldEnum;
+use Kanvas\Inventory\Regions\Models\Regions;
+use PHPShopify\ShopifySDK;
+
+class ShopifyService
+{
+    protected ShopifySDK $shopifySdk;
+
+    public function __construct(
+        protected AppInterface $app,
+        protected CompanyInterface $company,
+        protected Regions $region
+    ) {
+        $this->shopifySdk = Client::getInstance($app, $company, $region);
+    }
+
+    public static function shopifySetup(ShopifyDto $data): array
+    {
+        $clientKeyNaming = CustomFieldEnum::SHOPIFY_API_KEY->value ."-". $data->company->getId() ."-". $data->region->getId();
```

Comment:
@arfenis i would save it in 1 field as a json like you did with the S3 and google drive